kamelets/aws-s3-sink.kamelet.yaml

```diff
@@ -121,6 +121,12 @@ spec:
             - set-header:
                 name: CamelAwsS3Key
                 simple: "${header[ce-file]}"
+          - simple: "${properties:keyName:null} == 'null'"
+            steps:
+            - set-header:
+                name: CamelAwsS3Key
+                simple: "${exchangeId}"      
+                #
```

Comment:
I don"t think this line is needed

Reply:
I have added a comment there that was lost in the first commit.